code/modules/mob/mob_movement.dm

```diff
@@ -187,6 +187,9 @@
 		for(var/obj/O in mob.user_movement_hooks)
 			O.intercept_user_move(direct, mob, n, oldloc)
 
+	if(mob.pulling && !ismob(mob.pulling))

```

Comment:
Curious since this is adding a check to every movement, would it be better performance wise to have the typecheck nested, or does it not make a difference?

Reply:
No, the and operator is shortcircuited.
If the first is false, the second doesn't even gets checked.

Reply:
it doesn't make a difference. `if (foo && bar)` is equivalent to 
```
if (foo)
  if (bar)
``` 
You can convince yourself of this by looking at any place where the check goes like `if (foo && foo.somevar)`, where foo being null would cause a runtime in the later check